### PR TITLE
(WIP)chore: update default settings path 

### PR DESCRIPTION
**Pull request by swetar-mecha**
Update: Settings path to be automatically picked up
------------------------
- [x] status bar
- [x] settings panel  
- [x] lock screen
- [x] power options
- [x] notification
- [x] app switcher
- [x] app drawer
- [x] greeter
- [x] home screen
- [x] keyboard

### CODE REVIEW

File: shell/app-switcher/src/constants.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pub const SYSTEM_MECHANIX_APP_SWITCHER_PATH  :  &str  = "/usr/share/mechanix/app-switcher/icons/";
```
Comment (shoaibmerchant) [inline shell/app-switcher/src/constants.rs]: rename as 'ASSET_PATH' and it should be /usr/share/mechanix/app-switcher/assets/

File: shell/app-switcher/src/constants.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+pub const BECK_ICON : &str = "back.svg";
```
Comment (shoaibmerchant) [inline shell/app-switcher/src/constants.rs]: prefix _icon for all file names 'back_icon.svg'

File: shell/app-switcher/src/settings/mod.rs [hunk 3]
```diff
@@ -26,7 +28,7 @@ impl Default for AppSwitcherSettings {
             window: WindowSettings::default(),
             title: String::from("App Switcher"),
             modules: Modules::default(),
-            exclude_apps: vec![],
+            exclude_apps: vec!["mechanix.shell.home-screen".to_string()],
```
Comment (shoaibmerchant) [inline shell/app-switcher/src/settings/mod.rs]: move to constant.rs as 'APP_NAMESPACE'

File: shell/app-switcher/src/settings/mod.rs
Comment (shoaibmerchant) [inline shell/app-switcher/src/settings/mod.rs:1]: The settings precedence should be following

1. Passed as argument
2. Passed as env
3. Fetch from user's home directory `~/.config/mechanix/app-switcher/settings.yml`
4. Fallback to SETTINGS_PATH (declare in constants.rs) as `/usr/share/mechanix/app-switcher/settings.yml'

File: shell/app-switcher/src/settings/mod.rs [hunk 4]
```diff
@@ -92,30 +94,63 @@ pub struct LayoutSettings {
...
+#[derive(Debug, Deserialize, Clone, Serialize)]
+#[serde(default)]
 pub struct CloseAll {
-    pub icon: Option<String>,
+    pub icon: String,
+}
+impl Default for CloseAll {
+    fn default() -> Self {
+        Self { 
+           icon:  SYSTEM_MECHANIX_APP_SWITCHER_PATH.to_owned() + CLOSE_ALL_ICON,
```
Comment (akshayr-hub) [inline shell/app-switcher/src/settings/mod.rs]: can we do concatenation in constants file?

File: shell/app-switcher/Cargo.toml [hunk 5]
```diff
@@ -31,6 +31,8 @@ palette = "0.7.6"
 image = { workspace = true }
 colors-transform = "0.2.11"
 desktop_entries = { workspace = true }
+const_format = "0.2.32"
```
Comment (akshayr-hub) [inline shell/app-switcher/Cargo.toml]: as these are common among all shell components
you can add at workspace level shell/Cargo.toml
in  shell/Cargo.toml
const_format = "0.2.32"
in component
const_format = { workspace = true }

Comment (swetar-mecha) [reply shell/app-switcher/Cargo.toml]: okay

**Review by shoaibmerchant — APPROVED** [review]

**Pull request merged**